plugins/microsoft-entra-contractor-expiry-reminders/README.md

```diff
@@ -14,3 +13,45 @@ systems:
+
+
+
+# **Conversation Design**
+
+[This purple chat mock](https://developer.moveworks.com/creator-studio/developer-tools/purple-chat-builder/?workspace=%7B%22title%22%3A%22My+Workspace%22%2C%22botSettings%22%3A%7B%22name%22%3A%22%22%2C%22imageUrl%22%3A%22%22%7D%2C%22mocks%22%3A%5B%7B%22id%22%3A8851%2C%22title%22%3A%22New+Mock%22%2C%22transcript%22%3A%7B%22messages%22%3A%5B%7B%22from%22%3A%22BOT%22%2C%22text%22%3A%22%3Cp%3EHi+Andrew%2C%3Cbr%3E%3Cbr%3EThe+following+4+employees%27+contracts+are+expiring+soon.+Please+extend+or+terminate+each+of+the+following%3A%3Cbr%3E%3Cbr%3E1.+Anna+Harris+-+Expires+%3Cb%3EApril+30%2C+2023%3C%2Fb%3E%3Ca+href%3D%5C%22https%3A%2F%2Fcompany.contracts%2Fanna_harris%5C%22%3E+%28View+Contract%29%3C%2Fa%3E%3Cbr%3E2.+John+Doe+-+Expires+%3Cb%3EMay+15%2C+2023%3C%2Fb%3E%3Ca+href%3D%5C%22https%3A%2F%2Fcompany.contracts%2Fjohn_doe%5C%22%3E+%28View+Contract%29%3C%2Fa%3E%3Cbr%3E3.+Lisa+Ray+-+Expires+%3Cb%3EJune+1%2C+2023%3C%2Fb%3E%3Ca+href%3D%5C%22https%3A%2F%2Fcompany.contracts%2Flisa_ray%5C%22%3E+%28View+Contract%29%3C%2Fa%3E%3Cbr%3E4.+Mark+Smith+-+Expires+%3Cb%3EJuly+20%2C+2023%3C%2Fb%3E%3Ca+href%3D%5C%22https%3A%2F%2Fcompany.contracts%2Fmark_smith%5C%22%3E+%28View+Contract%29%3C%2Fa%3E%3Cbr%3E%3C%2Fp%3E%22%7D%5D%2C%22settings%22%3A%7B%22colorStyle%22%3A%22LIGHT%22%2C%22startTime%22%3A%2211%3A43%2BAM%22%2C%22defaultPerson%22%3A%22GWEN%22%2C%22editable%22%3Atrue%2C%22botName%22%3A%22%22%2C%22botImageUrl%22%3A%22%22%7D%7D%7D%5D%7D) shows the experience that can be achieved.
+
+# **Design Questions to Consider**
+
+1. What is the frequency of notification that you would like to send to the Managers for reminder?
+2. How would you like to send the reminder when multiple contractors under the same manager are expiring at once?
+
+# High-level overview of the Workflow
+
+The overall workflow would be a simple **Daily** job that does the following:
+
+- Queries Active Directory to get a list of User Accounts which are set to expire based on the start_date and end_date
+  - To do this, you have to use this filter: `(&(objectClass=user)(accountExpires>={start_date})(accountExpires<={end_date}))`
+  - and have the following attributes selected: `distinguishedName`, `name`, `mail`, `manager`, `accountExpires`
+  - The start date and end date will get incremented every day
+- You also need to track the notifications that were already sent out or else you might send multiple ones for the same account since we are dealing with a time window.
```

Comment:
Can we offer a way to do this without additional infrastructure dependencies? Maybe just send the notification every Monday if there's 90 days, 60 days, or 30 days remaining within an error margin of ±4 days.